src/rule_definition_tools.jl

```diff
@@ -418,27 +418,32 @@ function _nondiff_frule_expr(__source__, primal_sig_parts, primal_invoke)
     end
 end
 
-function tuple_expression(primal_sig_parts)
+function _make_pullback_for_non_differentiable(::Val{N}) where {N}
```

Comment:
It's a matter of taste, of course, but IMO a callable struct would be simpler and cleaner as the type of the function is cleaner and you don't have to deal with `Val`s.

Reply:
Should it subtype `Function`?